content/blog/2019/performance-best-practices/index.md

```diff
@@ -0,0 +1,110 @@
+
+Both of these exceptions will go away in a future Istio release, when [Mixer V2](https://docs.google.com/document/d/1QKmtem5jU_2F3Lh5SqLp0IuPb80_70J7aJEYu4_gS-s) moves all policy and telemetry features directly into the sidecar proxies.
+
+Next, when testing Istio's data plane performance at scale, it's important to test not only at increasing requests per second, but also against an increasing number of **concurrent** connections. This is because real-world, high-throughput traffic comes from multiple clients. The [provided scripts](https://github.com/istio/tools/tree/76e3cb2488303316c8511a3ebe9676828c9d4765/perf/benchmark#example-2) let you perform the same load test with any number of concurrent connections, at increasing RPS.
+
+Lastly, our test environment measures requests between two pods, not many. The client pod is [Fortio](http://fortio.org/), which sends traffic to the server pod.
+
+Why test with only two pods? Because scaling up throughput (RPS) and connections (threads) has a greater effect on Envoy's performance than increasing the total size of the service registry — or, the total number of pods and services in the Kubernetes cluster. When the size of the service registry grows, Envoy does have to keep track of more endpoints, and lookup time per request does increase, but by a tiny constant. If you have many services, and this constant becomes a latency concern, Istio provides a [Sidecar resource](/docs/reference/config/networking/v1alpha3/sidecar/), which allows you to limit which services each Envoy knows about.
+
+## 3. Measure with and without proxies
+
+While many Istio features, such as [mutual TLS authentication](/docs/concepts/security/#mutual-tls-authentication), rely on an Envoy proxy next to an application pod, you can [selectively disable](/docs/setup/kubernetes/additional-setup/sidecar-injection/#disabling-or-updating-the-webhook) sidecar proxy injection for some of your mesh services. As you scale up Istio for production, you may want to incrementally add the sidecar proxy to your workloads.
+
+To that end, the test scripts provide [three different modes](https://github.com/istio/tools/tree/de2ab3e4650a2eab47002928a42fd5616f395dc2/perf/benchmark#run-performance-tests). These modes analyze Istio's performance when a request goes through both the client and server proxies (`both`), just the server proxy (`serveronly`), and neither proxy (`baseline`). We also provide an option to disable [Mixer](/docs/concepts/observability/) (telemetry) during the performance tests, which will provide results in line with the performance we expect when the Mixer V2 work is completed.
```

Comment:
Add to this paragraph
```
Istio supports [Envoy native telemetry](https://github.com/istio/istio/wiki/Envoy-native-telemetry), and its performance is comparable to Mixer telemetry disabled.
```

Reply:
To simplify things, I removed a couple of Mixer mentions. I also tried to make the paragraph a bit easier to consume by turning it into a list and added @mandarjog suggestion, PTAL.
```suggestion
To that end, the test scripts provide [three different modes](https://github.com/istio/tools/tree/de2ab3e4650a2eab47002928a42fd5616f395dc2/perf/benchmark#run-performance-tests):

- Use `both` to analyze Istio's performance when a request goes through both the client and server proxies.
- Use `serveronly` to analyze Istio's performance when a request goes just through the server proxy
- Use `baseline` to analyze Istio's performance when a request goes through neither proxy.

We also provide an option to disable [telemetry](/docs/concepts/observability/) during the performance tests, which  provides results in line with the performance we expect when the Mixer V2 work is completed. Istio supports [Envoy native telemetry](https://github.com/istio/istio/wiki/Envoy-native-telemetry), and its performance is comparable to Istio's performance with telemetry disabled.
```